metadetect/lsst_metadetect.py

```diff
@@ -126,65 +133,47 @@ def _add_positions_and_psf(self, medsifier, res, shear_str):
                 peak = rec.getFootprint().getPeaks()[0]
                 orig_cen = peak.getI()
 
-            newres['row'][i] = orig_cen.getY()
-            newres['col'][i] = orig_cen.getX()
-            # print(newres['row'][i], newres['col'][i])
-        '''
-        newres['star_frac'][:] = self.star_frac
-        newres['tapebump_frac'][:] = self.tapebump_frac
-        newres['spline_interp_frac'][:] = self.spline_interp_frac
-        newres['noise_interp_frac'][:] = self.noise_interp_frac
-        newres['imperfect_frac'][:] = self.imperfect_frac
-
-        if cat.size > 0:
-            obs = self.mbobs[0][0]
+            newres['box_size'][i] = mbobs_list[i][0][0].image.shape[0]
+            newres['row0'][i] = bbox.getBeginY()
+            newres['col0'][i] = bbox.getBeginX()
+            newres['row'][i] = orig_cen.getY() - newres['row0'][i]
+            newres['col'][i] = orig_cen.getX() - newres['col0'][i]
+
+            newres['ormask'][i] = self.ormask[
+                int(newres['row'][i]), int(newres['col'][i]),
+            ]
+
+        if len(sources) > 0:
 
-            newres['sx_col'] = cat['x']
-            newres['sx_row'] = cat['y']
+            obs = self.mbobs[0][0]
 
             rows_noshear, cols_noshear = shearpos.unshear_positions(
-                newres['sx_row'],
-                newres['sx_col'],
+                newres['row'],
+                newres['col'],
                 shear_str,
                 obs,  # an example for jacobian and image shape
+                # default is 0.01 but make sure to use the passed in default
+                # if needed
+                step=self['metacal'].get("step", shearpos.DEFAULT_STEP),
             )
-
-            newres['sx_row_noshear'] = rows_noshear
-            newres['sx_col_noshear'] = cols_noshear
-
-            dims = obs.image.shape
-            rclip = _clip_and_round(rows_noshear, dims[0])
-            cclip = _clip_and_round(cols_noshear, dims[1])
-
-            if 'ormask_region' in self and self['ormask_region'] > 1:
```

Comment:
Did you mean to remove support for ormask_region?

Reply:
It was not implemented, and I don't remember what it was